packages/core/src/scanning.ts

```diff
@@ -109,3 +109,27 @@ export const scanOutputs = (
 
     return matches;
 };
+
+export const scanOutputsWithTweak = (
+    spendPublicKey: Buffer,
+    scanTweak: Buffer,
+    outputs: Buffer[],
+): Map<string, Buffer> => {
+    const matches = new Map<string, Buffer>();
+
+    const tweakedPublicKey = secp256k1.publicKeyTweakAdd(
+        spendPublicKey,
+        scanTweak,
+        true
+    );
+
+    for (let i = outputs.length - 1; i >= 0; i--) {
+        const output = outputs[i];
+        if (output.equals(tweakedPublicKey)) {
+            matches.set(output.toString('hex'), scanTweak);
+            outputs.splice(i, 1);
+        }
+    }
+
+    return matches;
+};
```

Comment:
@notTanveer  Understood,  kindly replace
 ``` typescript
const ecdhSecret = secp256k1.publicKeyTweakMul(
        sumOfInputPublicKeys,
        secp256k1.privateKeyTweakMul(scanPrivateKey, inputHash),
        true,
    );
```
with 
``` typescript
const ecdhSecret = secp256k1.privateKeyTweakMul(scanPrivateKey, scanTweak)
``` 
in this new function, but i believe everything else should be the same as in the scanOutputs function.

Reply:
in summary this function should be :
``` typescript
export const scanOutputsWithTweak = (
    scanPrivateKey: Buffer,
    spendPublicKey: Buffer,
    scanTweak: Buffer,
    outputs: Buffer[],
    labels?: LabelMap,
): Map<string, Buffer> => {
    const ecdhSecret = secp256k1.privateKeyTweakMul(scanPrivateKey, scanTweak)

    // output to tweak data map
    const matches = new Map<string, Buffer>();
    let n = 0;
    let counterIncrement = 0;

    do {
        const tweak = createTaggedHash(
            'BIP0352/SharedSecret',
            Buffer.concat([ecdhSecret, serialiseUint32(n)]),
        );
        counterIncrement = processTweak(
            spendPublicKey,
            tweak,
            outputs,
            matches,
            labels,
        );
        n += counterIncrement;
    } while (counterIncrement > 0 && outputs.length > 0);

    return matches;
};
```